scripts/babel-relay-plugin/src/GraphQLPrinter.js

```diff
@@ -499,9 +506,6 @@ function printField(
         ));
       }
     });
-    if (hasEdgesSelection && !!fieldDecl.type.getFields()['pageInfo']) {
-      subRequisiteFields.pageInfo = true;
-    }
   } else if (types.getNamedType(fieldDecl.type).name === 'PageInfo') {
```

Comment:
I thought about this a little. I believe it's unnecessary.

If you have a `fragment on PageInfo`, you must have a fragment or field on the connection above it, which already would have included the `pageInfo` field, and then pulled in `hasPreviousPage` and `hasNextPage` via this logic. (Also, whatever component has the `fragment on PageInfo` is unlikely to itself care anyway. Same holds with the cursor stuff.

The `edges`/`pageInfo` thing is special because it's a sibling requisite field rather than a child; that's what caused my initial issue.